tests/repository_simulator.py

```diff
@@ -184,6 +184,8 @@ def fetch(self, url: str) -> Iterator[bytes]:
             path = spliturl.path[len("/targets/") :]
             dir_parts, sep , prefixed_filename = path.rpartition("/")
             prefix, _, filename = prefixed_filename.partition(".")
+            if prefix == "":
+                raise ValueError("File name must have hash prefix")
```

Comment:
Isn't this error duplicating  `NotImplementedError("non-consistent snapshot not supported")`?

Reply:
I think it's not quite a duplicate, but it's not correct either. Our updaters expect to be able to download non-prefixed targets even if repo is consistent_snapshot: they even have a configuration option to do that... _Maybe_ that was a bad idea considering the unfortunate prefixed-filename-design but it is what Warehouse intends to use.

I don't think a repository can support _completely arbitrary_ target names with this prefix-design (if client asks for target "a.b" we have no way of really knowing if a is a hash or just part of non-prefixed filename, unless there's some external knowledge of how clients work)... but we just have to implement something in RepositorySimulator so we can test both prefixed and non-prefixed downloads.

Long term maybe there should actually be a `consistent_targets` root variable in the spec to solve this

Reply:
I don't think so, because if the prefix is "" here this means you have enabled consistent snapshot, but not used hash prefix.
Now when I think for this case this seems to be a possible valid situation - that's why we have the separate `prefix_targets_with_hash` variable, but also the repository simulator doesn't have access to it...
Wondering if I should remove that check considering that the repository simulator doesn't care about the value of `prefix_targets_with_hash` as long as consistent snapshot is True.

Reply:
> Long term maybe there should actually be a consistent_targets root variable in the spec to solve this
Yes, I like that suggestion. Multiple times I make the mistake to think that `consistent_targets` will be used only if `consistent_snapshot` is True. 
I think it will be beneficial if we split the idea of consistent targets and consistent snapshots.

The question remains: what should we do now? Maybe just remove that check as we don't have enough context to do it well?

Reply:
yeah the check is definitely not correct

Reply:
Removed the check and the exception.